x-pack/plugins/lens/public/shared_components/legend_size_settings.tsx

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { EuiFieldNumber, EuiFlexGroup, EuiFlexItem, EuiFormRow, EuiIconTip } from '@elastic/eui';
+import { TooltipWrapper } from './tooltip_wrapper';
+
+interface LegendSizeSettingsProps {
+  legendSize: number | undefined;
+  onLegendSizeChange: (size?: number) => void;
+  isDisabled: boolean;
+}
+
+export const LegendSizeSettings = ({
+  legendSize,
+  onLegendSizeChange,
+  isDisabled,
+}: LegendSizeSettingsProps) => (
+  <EuiFormRow
+    display="columnCompressed"
+    label={i18n.translate('xpack.lens.shared.legendSizeSetting.label', {
+      defaultMessage: 'Legend size',
+    })}
+  >
+    <TooltipWrapper
+      tooltipContent={i18n.translate('xpack.lens.shared.legendVisibleTooltip', {
+        defaultMessage: 'Requires legend to be shown',
+      })}
+      condition={isDisabled}
+      position="top"
+      delay="regular"
+      display="block"
+    >
+      <EuiFlexGroup gutterSize="s" alignItems="center" responsive={false}>
+        <EuiFlexItem>
+          <EuiFieldNumber
+            placeholder={i18n.translate('xpack.lens.shared.legendSizeSetting.placeholder', {
+              defaultMessage: 'Auto',
+            })}
+            value={legendSize}
+            min={1}
+            step={1}
+            compressed
+            disabled={isDisabled}
+            onChange={(e) => {
+              const value = Number(e.target.value) || undefined;
+              onLegendSizeChange(value);
+            }}
+            append={i18n.translate('xpack.lens.shared.legendSizeSetting.fieldAppend', {
+              defaultMessage: 'px',
+            })}
+          />
```

Comment:
> 1) change the input type from pixels to percentages
@MichaelMarcialis I don't have a strong opinion here but I can list two reasons why we should not do that:
1) at a dashboard level you can't nicely align the legend of multiple charts vertically if each chart has different sizes
2) the legend content doesn't resize/grow/expand in percentage as the legend. Adding configuring it in percentage means that if I configured my chart in lens (with the lens ~4/3 aspect ratio) i probably get a different aspect ratio in a dashboard. If I configured a 20% legend width in a 500px width time chart in Lens and then I'm stretching my chart to the full dashboard horizontal space (let's say 1920px), I may end up with a legend ~400px wide that is actually a bit too much. 

It could be instead interesting to change, as a first iteration, the input to a selector with a bunch of predefined sizes like: small (50px), medium (100px), large (200px) xl (300px) etc so it's easier for the user to pick the same choice on multiple charts

Reply:
> 2) change this number input to an EuiRange slider (with accompanying showInput prop)

I agree that slider could fit perfectly, but unfortunately, we can't tell the chart size from settings bar as it configures the chart without accessing its actual properties like size. That is why we have to go with that limits explanatory tooltip

Reply:
> It could be instead interesting to change, as a first iteration, the input to a selector with a bunch of predefined sizes like: small (50px), medium (100px), large (200px) xl (300px) etc so it's easier for the user to pick the same choice on multiple charts

We've adopted this strategy multiple times and I think it makes sense here.

Reply:
Also, such preset-based interface will reduce a bit the current legend popover layout, which is starting to be quite articulated:

<img width="359" alt="Screenshot 2022-02-24 at 16 43 08" src="https://user-images.githubusercontent.com/924948/155557556-bcf8584b-c584-4bc0-81a0-3f7a44d15823.png">

Reply:
> 1. at a dashboard level you can't nicely align the legend of multiple charts vertically if each chart has different sizes
> 2. the legend content doesn't resize/grow/expand in percentage as the legend. Adding configuring it in percentage means that if I configured my chart in lens (with the lens ~4/3 aspect ratio) i probably get a different aspect ratio in a dashboard. If I configured a 20% legend width in a 500px width time chart in Lens and then I'm stretching my chart to the full dashboard horizontal space (let's say 1920px), I may end up with a legend ~400px wide that is actually a bit too much.

@markov00: Excellent points. I spoke too soon and wasn't considering the repercussions this would have on dashboards.

> It could be instead interesting to change, as a first iteration, the input to a selector with a bunch of predefined sizes like: small (50px), medium (100px), large (200px) xl (300px) etc so it's easier for the user to pick the same choice on multiple charts

@markov00, @dej611: I really like this suggestion for how easy it makes it for users to pick a consistent size and because it could potentially alleviate the need for us to even have an "auto" mode at all. Instead, we could simply default to whichever of these predefined sizes makes the most sense. Or do we still require an "auto" sizing mode to allow the legend to collapse down to the width/height of the legend's contents?

In any case, I think we can consolidate the auto toggle and legend size rows into a single row with either an `EuiSelect` or `EuiButtonGroup` that contains the available size options (including auto if it is still necessary). Would you be open to making such a change, @DziyanaDzeraviankina?

Reply:
@MichaelMarcialis the `auto` legend size was a feature requested or preexisting in vislib, it has its own pros and cons but it still an opportunity. I'm totally happy to remove it (since it will alleviate our constraints and layout calculation) but I think this deserve discussing a bit more. Can we chat on Monday about that?